Library/Homebrew/utils/ruby_check_version_script.rb

```diff
@@ -17,7 +17,8 @@
 homebrew_required_ruby_version_major, homebrew_required_ruby_version_minor, =
   homebrew_required_ruby_version.canonical_segments
 
-if ruby_version_major != homebrew_required_ruby_version_major ||
-   ruby_version_minor != homebrew_required_ruby_version_minor
+if (ENV["HOMEBREW_DEVELOPER"] == "1" && homebrew_required_ruby_version > ruby_version) ||
```

Comment:
```suggestion
if (ENV["HOMEBREW_DEVELOPER"].present? && homebrew_required_ruby_version > ruby_version) ||
```
Per the docs you should be able to set this to anything other than the empty string.

Unless we want to gate this specifically to users who have this set to `1`, that is.